--- nurbs/Makefile.am.org	Thu Aug  5 03:49:16 1999
+++ nurbs/Makefile.am	Tue Mar 13 21:37:05 2001
@@ -1,12 +1,12 @@
-EXTRA_DIST = nurbs.cc curve.cc hnurbs.cc hnurbsS.cc hnurbsS_sp.cc matrixRT.cc nurbs.cc nurbsArray.cc nurbsGL.cc nurbsS.cc nurbsS_sp.cc nurbsSub.cc nurbs_sp.cc surface.cc tri_spline.cc
+EXTRA_DIST = nurbs.cc curve.cc hnurbs.cc hnurbsS.cc hnurbsS_sp.cc matrixRT.cc nurbs.cc nurbsArray.cc nurbsS.cc nurbsS_sp.cc nurbsSub.cc nurbs_sp.cc surface.cc tri_spline.cc
 
 MAGICK_INCLUDE =  $(X_CFLAGS) @magick_include@
 
 INCLUDES = $(MAGICK_INCLUDE) -I@top_srcdir@/include @matrix_include@ $(GL_CFLAGS)
 
-pkginclude_HEADERS = curve.hh matrixRT.hh nurbsSub.hh hnurbs.hh nurbs.hh nurbs_sp.hh hnurbsS.hh nurbsGL.hh surface.hh hnurbsS_sp.hh nurbsS.hh nurbsS_sp.hh tri_spline.hh nurbs_global.hh
+pkginclude_HEADERS = curve.hh matrixRT.hh nurbsSub.hh hnurbs.hh nurbs.hh nurbs_sp.hh hnurbsS.hh surface.hh hnurbsS_sp.hh nurbsS.hh nurbsS_sp.hh tri_spline.hh nurbs_global.hh
 
-float_sources = f_nurbs.cc f_nurbsS.cc f_curve.cc f_surface.cc f_hnurbsS.cc matrixRT.cc f_nurbs_sp.cc f_nurbsS_sp.cc f_hnurbsS_sp.cc nurbsGL.cc f_nurbsArray.cc f_nurbsSub.cc f_tri_spline.cc
+float_sources = f_nurbs.cc f_nurbsS.cc f_curve.cc f_surface.cc f_hnurbsS.cc matrixRT.cc f_nurbs_sp.cc f_nurbsS_sp.cc f_hnurbsS_sp.cc f_nurbsArray.cc f_nurbsSub.cc f_tri_spline.cc
 
 double_sources = d_nurbs.cc d_nurbsS.cc d_curve.cc d_surface.cc d_hnurbsS.cc matrixRT.cc d_nurbs_sp.cc d_nurbsS_sp.cc d_hnurbsS_sp.cc d_nurbsArray.cc  d_nurbsSub.cc d_tri_spline.cc
 
@@ -16,7 +16,7 @@
 #lib_LTLIBRARIES = @build_lib@
 
 
-#libnurbsf_la_SOURCES = f_nurbs.cc f_nurbsS.cc f_curve.cc f_surface.cc f_hnurbsS.cc matrixRT.cc f_nurbs_sp.cc f_nurbsS_sp.cc f_hnurbsS_sp.cc nurbsGL.cc f_nurbsArray.cc f_nurbsSub.cc
+#libnurbsf_la_SOURCES = f_nurbs.cc f_nurbsS.cc f_curve.cc f_surface.cc f_hnurbsS.cc matrixRT.cc f_nurbs_sp.cc f_nurbsS_sp.cc f_hnurbsS_sp.cc f_nurbsArray.cc f_nurbsSub.cc
 
 #libnurbsd_la_SOURCES = d_nurbs.cc d_nurbsS.cc d_curve.cc d_surface.cc d_hnurbsS.cc matrixRT.cc d_nurbs_sp.cc d_nurbsS_sp.cc d_hnurbsS_sp.cc d_nurbsArray.cc  d_nurbsSub.cc
 
--- nurbs/Makefile.in.org	Mon Sep 13 02:59:03 1999
+++ nurbs/Makefile.in	Tue Mar 13 21:37:44 2001
@@ -105,15 +105,15 @@
 perl = @perl@
 prefix = @prefix@
 
-EXTRA_DIST = nurbs.cc curve.cc hnurbs.cc hnurbsS.cc hnurbsS_sp.cc matrixRT.cc nurbs.cc nurbsArray.cc nurbsGL.cc nurbsS.cc nurbsS_sp.cc nurbsSub.cc nurbs_sp.cc surface.cc tri_spline.cc
+EXTRA_DIST = nurbs.cc curve.cc hnurbs.cc hnurbsS.cc hnurbsS_sp.cc matrixRT.cc nurbs.cc nurbsArray.cc nurbsS.cc nurbsS_sp.cc nurbsSub.cc nurbs_sp.cc surface.cc tri_spline.cc
 
 MAGICK_INCLUDE = $(X_CFLAGS) @magick_include@
 
 INCLUDES = $(MAGICK_INCLUDE) -I@top_srcdir@/include @matrix_include@ $(GL_CFLAGS)
 
-pkginclude_HEADERS = curve.hh matrixRT.hh nurbsSub.hh hnurbs.hh nurbs.hh nurbs_sp.hh hnurbsS.hh nurbsGL.hh surface.hh hnurbsS_sp.hh nurbsS.hh nurbsS_sp.hh tri_spline.hh nurbs_global.hh
+pkginclude_HEADERS = curve.hh matrixRT.hh nurbsSub.hh hnurbs.hh nurbs.hh nurbs_sp.hh hnurbsS.hh surface.hh hnurbsS_sp.hh nurbsS.hh nurbsS_sp.hh tri_spline.hh nurbs_global.hh
 
-float_sources = f_nurbs.cc f_nurbsS.cc f_curve.cc f_surface.cc f_hnurbsS.cc matrixRT.cc f_nurbs_sp.cc f_nurbsS_sp.cc f_hnurbsS_sp.cc nurbsGL.cc f_nurbsArray.cc f_nurbsSub.cc f_tri_spline.cc
+float_sources = f_nurbs.cc f_nurbsS.cc f_curve.cc f_surface.cc f_hnurbsS.cc matrixRT.cc f_nurbs_sp.cc f_nurbsS_sp.cc f_hnurbsS_sp.cc f_nurbsArray.cc f_nurbsSub.cc f_tri_spline.cc
 
 double_sources = d_nurbs.cc d_nurbsS.cc d_curve.cc d_surface.cc d_hnurbsS.cc matrixRT.cc d_nurbs_sp.cc d_nurbsS_sp.cc d_hnurbsS_sp.cc d_nurbsArray.cc  d_nurbsSub.cc d_tri_spline.cc
 
@@ -122,7 +122,7 @@
 #EXTRA_LTLIBRARIES = libnurbsf.la libnurbsd.la
 #lib_LTLIBRARIES = @build_lib@
 
-#libnurbsf_la_SOURCES = f_nurbs.cc f_nurbsS.cc f_curve.cc f_surface.cc f_hnurbsS.cc matrixRT.cc f_nurbs_sp.cc f_nurbsS_sp.cc f_hnurbsS_sp.cc nurbsGL.cc f_nurbsArray.cc f_nurbsSub.cc
+#libnurbsf_la_SOURCES = f_nurbs.cc f_nurbsS.cc f_curve.cc f_surface.cc f_hnurbsS.cc matrixRT.cc f_nurbs_sp.cc f_nurbsS_sp.cc f_hnurbsS_sp.cc f_nurbsArray.cc f_nurbsSub.cc
 
 #libnurbsd_la_SOURCES = d_nurbs.cc d_nurbsS.cc d_curve.cc d_surface.cc d_hnurbsS.cc matrixRT.cc d_nurbs_sp.cc d_nurbsS_sp.cc d_hnurbsS_sp.cc d_nurbsArray.cc  d_nurbsSub.cc
 
@@ -170,7 +170,7 @@
 libnurbsf_la_LIBADD = 
 libnurbsf_la_OBJECTS =  f_nurbs.lo f_nurbsS.lo f_curve.lo f_surface.lo \
 f_hnurbsS.lo matrixRT.lo f_nurbs_sp.lo f_nurbsS_sp.lo f_hnurbsS_sp.lo \
-nurbsGL.lo f_nurbsArray.lo f_nurbsSub.lo f_tri_spline.lo
+f_nurbsArray.lo f_nurbsSub.lo f_tri_spline.lo
 libnurbsd_la_LIBADD = 
 libnurbsd_la_OBJECTS =  d_nurbs.lo d_nurbsS.lo d_curve.lo d_surface.lo \
 d_hnurbsS.lo matrixRT.lo d_nurbs_sp.lo d_nurbsS_sp.lo d_hnurbsS_sp.lo \
@@ -561,14 +561,6 @@
 	../matrix/hpoint_nd.hh ../matrix/galloc2d.hh \
 	../matrix/vector.hh ../matrix/barray.hh ../matrix/list.hh \
 	../matrix/galloc.hh
-nurbsGL.lo nurbsGL.o : nurbsGL.cc matrixRT.hh nurbs_global.hh \
-	../matrix/matrix_global.hh ../include/plib_config.hh \
-	../matrix/error.hh ../matrix/matrix.hh ../matrix/barray2d.hh \
-	../matrix/specialType.hh ../matrix/matrixTool.hh \
-	../matrix/coordinate.hh ../matrix/point_nd.hh \
-	../matrix/hpoint_nd.hh ../matrix/galloc2d.hh \
-	../matrix/vector.hh ../matrix/barray.hh ../matrix/list.hh \
-	../matrix/galloc.hh nurbsGL.hh
 
 info-am:
 info: info-am
