--- nurbs++-3.0.11/nurbs/nurbsGL.cpp.orig	Fri Jun  6 14:41:42 2003
+++ nurbs++-3.0.11/nurbs/nurbsGL.cpp	Fri Jun  6 14:47:56 2003
@@ -3919,7 +3919,7 @@
 
 template class std::allocator<PLib::NurbsCurve<float, 2> *>;
 
-template void  std::_List_base<PLib::NurbsCurve<float, 2> *, std::allocator<PLib::NurbsCurve<float, 2> *> >::clear(void);
+template void std::_List_base<PLib::NurbsCurve<float, 2> *, std::allocator<PLib::NurbsCurve<float, 2> *> >::__clear(void);
 
 template void std::list<NurbsCurve_2Df*>::clear(void);
 
