diff -Nur nurbs++-3.0.11.orig/config/has_opengl.m4 nurbs++-3.0.11/config/has_opengl.m4
--- nurbs++-3.0.11.orig/config/has_opengl.m4	Fri May 17 18:03:18 2002
+++ nurbs++-3.0.11/config/has_opengl.m4	Tue Jul  9 23:16:08 2002
@@ -163,7 +163,7 @@
         else
             GL_LFLAGS="-L${ac_cv_with_gl_lib}"
         fi
-	GL_LIBS="-lGLU -lGL $X_PRE_LIBS -lXext -lm"
+	GL_LIBS="-L/usr/X11R6/lib -lGLU -lGL $X_PRE_LIBS -lXext -lm"
         AC_SUBST(GL_LIBS)
 	AC_SUBST(GL_LFLAGS)
     ],
@@ -185,7 +185,7 @@
         else
             GL_LFLAGS="-L${ac_cv_with_gl_lib}"
         fi
-	GL_LIBS="-lMesaGLU -lMesaGL -lXext -lm"
+	GL_LIBS="-L/usr/X11R6/lib -lMesaGLU -lMesaGL -lXext -lm"
 	AC_SUBST(GL_LFLAGS)
         AC_SUBST(GL_LIBS)
     ],
diff -Nur nurbs++-3.0.11.orig/image/Makefile.am nurbs++-3.0.11/image/Makefile.am
--- nurbs++-3.0.11.orig/image/Makefile.am	Mon May 13 22:04:34 2002
+++ nurbs++-3.0.11/image/Makefile.am	Tue Jul  9 23:20:17 2002
@@ -9,4 +9,5 @@
 
 lib_LTLIBRARIES = libmatrixI.la
 libmatrixI_la_SOURCES = color.cpp image_.cpp  rec_filter_.cpp filter_.cpp
-libmatrixI_la_LDFLAGS = -version-info 1:0:0
+libmatrixI_la_LDFLAGS = -version-info 1:0:0 $(MAGICK_LDFLAGS)
+libmatrixI_la_LIBADD = -L../matrix -lmatrix $(MAGICK_LIBS)
diff -Nur nurbs++-3.0.11.orig/numerical/Makefile.am nurbs++-3.0.11/numerical/Makefile.am
--- nurbs++-3.0.11.orig/numerical/Makefile.am	Mon May 13 22:04:38 2002
+++ nurbs++-3.0.11/numerical/Makefile.am	Tue Jul  9 23:20:39 2002
@@ -7,4 +7,4 @@
 lib_LTLIBRARIES = libmatrixN.la
 libmatrixN_la_SOURCES = matrixMat_.cpp fft_.cpp chebexp_.cpp intccq_.cpp statistic_.cpp
 libmatrixN_la_LDFLAGS = -version-info 1:0:0
-
+libmatrixN_la_LIBADD = -L../matrix -lmatrix
diff -Nur nurbs++-3.0.11.orig/nurbs/Makefile.am nurbs++-3.0.11/nurbs/Makefile.am
--- nurbs++-3.0.11.orig/nurbs/Makefile.am	Mon May 13 22:04:42 2002
+++ nurbs++-3.0.11/nurbs/Makefile.am	Tue Jul  9 23:22:56 2002
@@ -45,6 +45,7 @@
 lib_LTLIBRARIES = libnurbsf.la libnurbsd.la
 libnurbsf_la_SOURCES = $(float_sources)
 libnurbsf_la_LDFLAGS = $(ldflags)
+libnurbsf_la_LIBADD = -L../matrix -lmatrix -L../image -lmatrixI -L../numerical -lmatrixN $(GL_LIBS)
 libnurbsd_la_SOURCES = $(double_sources)
 libnurbsd_la_LDFLAGS = $(ldflags)
-
+libnurbsd_la_LIBADD = -L../matrix -lmatrix -L../image -lmatrixI -L../numerical -lmatrixN
