diff -Nur nurbs++-3.0.11.orig/nurbs/nurbsGL.cpp nurbs++-3.0.11/nurbs/nurbsGL.cpp
--- nurbs++-3.0.11.orig/nurbs/nurbsGL.cpp	Mon May 13 23:07:46 2002
+++ nurbs++-3.0.11/nurbs/nurbsGL.cpp	Tue Jul  9 22:54:22 2002
@@ -3913,7 +3913,7 @@
 #ifdef NO_IMPLICIT_TEMPLATES
 
 template class std::list<NurbsCurve_2Df*> ;
-template class std::list<NurbsCurve_2Df*>::iterator ;
+template  std::list<NurbsCurve_2Df*>::iterator ;
 
 template class RenderMeshGL<float> ;
 
